clevis.go

```diff
@@ -101,23 +99,38 @@ func (p Pin) ToConfig() (Config, error) {
 
 // Decrypt decrypts a clevis bound message. The message format can be either compact or JSON.
 func Decrypt(data []byte) ([]byte, error) {
-	msg, p, err := Parse(data)
+	return jwe.Decrypt(data, "", nil, jwe.WithPostParser(clevisParser{}))
+}
+
+type clevisParser struct{}
+
+// Do implements a jwe.PostParser interface which must use the excracted pin config to set up the decryprion algorithm and key
+func (clevisParser) Do(ctx jwe.DecryptCtx) error {
+	p, err := pinFromMsg(ctx.Message())
 	if err != nil {
-		return nil, err
+		return err
 	}
 
+	var key []byte
 	switch p.Pin {
```

Comment:
move this `switch` to a separate function `Pin.recoverKey(message)` maybe?

Reply:
I was thinking about something like this https://github.com/anatol/clevis.go/commit/7bd0e7bb7c46223088dbf3239d4492363c8fcd66

If you are ok with it ^^ then feel free to squash it into your change and push it to the PR.